tests/tck-build-logic/src/main/java/org/graalvm/internal/tck/ContributionTask.java

```diff
@@ -0,0 +1,453 @@
+package org.graalvm.internal.tck;
+
+import com.fasterxml.jackson.annotation.JsonInclude;
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.core.util.DefaultIndenter;
+import com.fasterxml.jackson.core.util.DefaultPrettyPrinter;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.SerializationFeature;
+import org.graalvm.internal.tck.exceptions.ContributingException;
+import org.graalvm.internal.tck.model.MetadataIndexEntry;
+import org.graalvm.internal.tck.model.contributing.Question;
+import org.graalvm.internal.tck.utils.ConfigurationStringBuilder;
+import org.graalvm.internal.tck.utils.FilesUtils;
+import org.graalvm.internal.tck.utils.InteractiveTaskUtils;
+import org.gradle.api.DefaultTask;
+import org.gradle.api.file.FileSystemOperations;
+import org.gradle.api.tasks.TaskAction;
+import org.gradle.process.ExecOperations;
+
+import javax.inject.Inject;
+import java.io.ByteArrayOutputStream;
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.StandardOpenOption;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Comparator;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+
+public abstract class ContributionTask extends DefaultTask {
+    private static final String BRANCH_NAME_PREFIX = "add-support-for-";
+    private static final String METADATA_INDEX = "metadata/index.json";
+    private static final String BUILD_FILE = "build.gradle";
+    private static final String USER_CODE_FILTER_FILE = "user-code-filter.json";
+    private static final String REQUIRED_DOCKER_IMAGES_FILE = "required-docker-images.txt";
+
+    @Inject
+    protected abstract ExecOperations getExecOperations();
+
+    @Inject
+    protected abstract FileSystemOperations getFileSystemOperations();
+
+    private final ObjectMapper objectMapper = new ObjectMapper().enable(SerializationFeature.INDENT_OUTPUT).setSerializationInclusion(JsonInclude.Include.NON_NULL);
+
+    private Path gradlew;
+    private Path testsDirectory;
+    private Path metadataDirectory;
+
+    private Coordinates coordinates;
+
+    private record ContributingQuestion(String question, String help) {}
+    private final Map<String, ContributingQuestion> questions = new HashMap<>();
+
+    private void initializeWorkingDirectories(){
+        testsDirectory = Path.of(getProject().file(CoordinateUtils.replace("tests/src/$group$/$artifact$/$version$", coordinates)).getAbsolutePath());
```

Comment:
This uses `getProject()` at task execution time. Be aware that this is deprecated (because of configuration cache) so will not work anymore at some point in time. Instead, you should use the project's layout directory.